From 4eade711314c4f8507e69213b1bf49fbba01b6d0 Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Wed, 5 Mar 2025 10:31:50 +0000
Subject: [PATCH] DT: bcm2712: swap PCIe QoS properties to the new array type

pcie1 should use the FIFO threshold property as the RC should not
pay attention to Vendor Messages from incompatible endpoint hardware.

Also drop the downstream MPS property, it's no longer needed.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi      | 9 +++++++++
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts  | 6 ------
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi | 6 ------
 3 files changed, 9 insertions(+), 12 deletions(-)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
@@ -674,3 +674,12 @@
 &vc4 {
 	status = "disabled";
 };
+
+&pcie1 {
+	brcm,fifo-qos-map = /bits/ 8 <3 3 3 3>;
+	status = "disabled";
+};
+
+&pcie2 {
+	brcm,vdm-qos-map = /bits/ 8 <8 8 8 9 10 10 11 11>;
+};
\ No newline at end of file
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -127,16 +127,10 @@
 };
 
 rp1_target: &pcie2 {
-	brcm,enable-mps-rcb;
-	brcm,vdm-qos-map = <0xbbaa9888>;
 	aspm-no-l0s;
 	status = "okay";
 };
 
-&pcie1 {
-	brcm,vdm-qos-map = <0x33333333>;
-};
-
 // Add some labels to 2712 device
 
 // The system UART
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -112,16 +112,10 @@
 };
 
 rp1_target: &pcie2 {
-	brcm,enable-mps-rcb;
-	brcm,vdm-qos-map = <0xbbaa9888>;
 	aspm-no-l0s;
 	status = "okay";
 };
 
-&pcie1 {
-	brcm,vdm-qos-map = <0x33333333>;
-};
-
 // Add some labels to 2712 device
 
 // The system UART
